ui.apps/src/main/javascript/minicart/src/components/Minicart/product.js

```diff
@@ -0,0 +1,92 @@
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+import React, { useState, useMemo, useCallback } from 'react';
+import { func, number, shape, object } from 'prop-types';
+import { Price } from '@magento/peregrine';
+import classes from './product.css';
+
+import { transparentPlaceholder } from '../../utils/transparentPlaceholder';
+import makeUrl from '../../utils/makeUrl';
+import Kebab from './kebab';
+import Section from './section';
+import { string } from 'postcss-selector-parser';
+
+const imageWidth = 80;
+const imageHeight = 100;
+
+const Product = props => {
+    const { beginEditItem, item, removeItemFromCart } = props;
+
+    const { product = undefined, quantity = 0, id = '' } = item;
+    const { thumbnail, name, price } = product;
+    const { value, currency } = price.regularPrice.amount;
+
+    console.log(`Product  is`, product);
+    const [isLoading, setIsLoading] = useState(false);
+
+    const productImage = useMemo(() => {
+        const src =
+            thumbnail && thumbnail.url
+                ? makeUrl(thumbnail.url, { type: 'image-product', width: imageWidth, height: imageHeight })
+                : transparentPlaceholder;
+        return <img alt={name} className={classes.image} placeholder={transparentPlaceholder} src={src} />;
+    });
+
+    const handleEditItem = useCallback(() => {
+        beginEditItem(item);
+    }, [beginEditItem, item]);
+
+    const handleRemoveItem = useCallback(() => {
+        setIsLoading(true);
+        removeItemFromCart(id);
+    }, [setIsLoading, id, removeItemFromCart]);
+
+    const mask = isLoading ? <div className={classes.mask} /> : null;
+    return (
+        <li className={classes.root}>
+            {productImage}
+            <div className={classes.name}>{name}</div>
+            <div className={classes.quantity}>
+                <div className={classes.quantityRow}>
+                    <span>{quantity}</span>
+                    <span className={classes.quantityOperator}>{'×'}</span>
+                    <span className={classes.price}>
+                        <Price currencyCode={currency} value={value} />
+                    </span>
+                </div>
+            </div>
+            {mask}
+            <Kebab>
+                <Section text="Edit item" onClick={handleEditItem} icon="Edit2" />
+                <Section text="Remove item" onClick={handleRemoveItem} icon="Trash" />
+            </Kebab>
+        </li>
+    );
+};
+
+Product.propTypes = {
+    removeItemFromCart: func.isRequired,
+    beginEditItem: func.isRequired,
+    item: shape({
+        id: string.isRequired,
+        quantity: number.isRequired,
+        product: shape({
+            name: string.isRequired,
+            price: object.isRequired,
```

Comment:
Looking at line 33 `const { value, currency } = price.regularPrice.amount;`, you may want to expand this and make `regularPrice` and `amount` required too.